### ISSUE TITLE
Provide native collapse all / expand all for TreeViews created by `registerTreeDataProvider`

### ISSUE THREAD

**Issue by christian-bromann** [microsoft/vscode#172479]
I had a simple requirement to build a custom TreeViewer in the explorer panel that has a collapse/expand all button, here is what I came up with:
![tree](https://user-images.githubusercontent.com/731337/214711742-c864007b-74d0-4514-85a9-52a9ccf65b37.gif)

Implementing this exposed two issues with VS Code and its API:

- I discovered that using `createTreeView` allows passing in `TreeViewOptions` which has a `showCollapseAll`, unfortunately this option can't be set when using `registerTreeDataProvider` and it doesn't contain an option to have an expand all either
- of course you can implement this yourself, the problem with that is that `TreeItem`s don't (e.g. updates to `collapsibleState` property) update if the label doesn't change, because of this I have to attach a space character to it so that VS Code triggers re-rendering it

### Proposal

To me it seems like collapse/expand all is a common scenario that could be handled by VS Codes API. Therefor I would suggest to allow setting a property like `showCollapseAll` and `showExpandAll` to the `TreeDataProvider`. By default both is set to `false`. A developer could set these to `true` in the constructor. If both are set to `true` VS Code uses a single button representing both (as demonstrated in my demo above).

There surely is more detail to this but before moving on, I would like to gauge whether I am completely wrong here and if my suggestion is something the VS Code team would accept. I am happy to contribute this myself.

**Comment by alexr00** [comment]
@christian-bromann `registerTreeDataProvider` is intended to be only for registering a data provider. If you want to control view related things (like having a collapse all button) then `createTreeView` is the way to go. Is there some reason why you can't use `createTreeView`?

**Comment by christian-bromann** [comment]
> Is there some reason why you can't use `createTreeView`?

Not really. However it only solves half of the problem because I would need an expand all button too. I see that the file tree has only a collapse all button (I assume because expanding all files wouldn't make much sense) but the outline has an expand/collapse all as shown above.

**Comment by alexr00** [comment]
In that case, the feature request for expand all is here: https://github.com/microsoft/vscode/issues/170636.

**Comment by christian-bromann** [comment]
Thank you!
